crypto/bio/bss_dgram.c

```diff
@@ -1574,17 +1588,14 @@ static long dgram_sctp_ctrl(BIO *b, int cmd, long num, void *ptr)
 }
 
 int BIO_dgram_sctp_notification_cb(BIO *b,
-                                   void (*handle_notifications) (BIO *bio,
-                                                                 void
-                                                                 *context,
-                                                                 void *buf),
-                                   void *context)
+                BIO_dgram_sctp_notification_handler_fn handle_notifications,
+                void *context)
 {
-    bio_dgram_sctp_data *data = (bio_dgram_sctp_data *) b->ptr;
-
     if (handle_notifications != NULL) {
-        data->handle_notifications = handle_notifications;
-        data->notification_context = context;
+        BIO_ctrl(b, BIO_CTRL_DGRAM_SCTP_SET_NOTIFICATION_HANDLER, 0,
+                 handle_notifications);
```

Comment:
I seem to recall that there are problems with passing a function pointer via a "void *" in some compilers.

@openssl - can anyone remind me of the issue?

Reply:
Pedantically, a ```void *``` need not be able to hold a function pointer.

Reply:
C89 doesn't allow it.

> A pointer to void may be converted to or from a pointer to any incomplete or object type. A pointer to any incomplete or object type may be converted to a pointer to void and back again: the result shall compare equal to the original pointer.

C89 distinguishes pointers to functions from pointers to incomplete or object types, so it is implied.

Reply:
Thanks for informing this. Changed the code which passes function pointer as `void *`.